jgrapht-core/src/main/java/org/jgrapht/alg/cycle/AhujaOrlinSharmaCyclicExchangeLocalAugmentation.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * (C) Copyright 2003-2018, by Christoph Grüne and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.util.TypeUtil;
+
+import java.util.*;
+
+/**
+ * Implementation of an algorithm for the local augmentation problem for the cyclic exchange neighborhood,
+ * i.e. it finds subset-disjoint negative cycles in a graph, based on
+ * Ravindra K. Ahuja, James B. Orlin, Dushyant Sharma,
+ * A composite very large-scale neighborhood structure for the capacitated minimum spanning tree problem,
+ * Operations Research Letters, Volume 31, Issue 3, 2003, Pages 185-194, ISSN 0167-6377,
+ * https://doi.org/10.1016/S0167-6377(02)00236-5. (http://www.sciencedirect.com/science/article/pii/S0167637702002365)
+ *
+ * This algorithm may enumerate all paths up to the length given by the parameter <code>lengthBound</code>,
+ * i.e the algorithm runs in exponential time.
+ *
+ * @param <V> the vertex type
+ * @param <E> the edge type
+ *
+ * @author Christoph Grüne
+ * @since June 7, 2018
+ */
+public class AhujaOrlinSharmaCyclicExchangeLocalAugmentation<V, E> {
+
+    /**
+     * Implementation of a labeled path.
+     * It is used in AhujaOrlinSharmaCyclicExchangeLocalAugmentation to efficiently maintain the paths in the calculation.
+     *
+     * @param <V> the vertex type
+     *
+     * @author Christoph Grüne
+     * @since June 7, 2018
+     */
+    public static class LabeledPath<V> implements Cloneable {
+
+        private LinkedList<V> vertices;
+        private Map<V, Integer> labels;
```

Comment:
I believe this path-level map is always a submap of the graph-level labels map (i.e. we never override labels when constructing a path).  If that's true, wouldn't it be simpler to just access the graph-level map (which you can do by making LabeledPath non-static)?